src/industry_gui.cpp

```diff
@@ -1430,12 +1440,26 @@ class IndustryDirectoryWindow : public Window {
 	 */
 	static int GetCargoTransportedSortValue(const Industry *i)
 	{
-		int p1 = GetCargoTransportedPercentsIfValid(i, 0);
-		int p2 = GetCargoTransportedPercentsIfValid(i, 1);
-
-		if (p1 > p2) Swap(p1, p2); // lower value has higher priority
+		CargoID filter = IndustryDirectoryWindow::produced_cargo_filter;
+
+		int percentage = 0, produced_cargo_count = 0;
+		for (uint id = 0; id < lengthof(i->produced_cargo); id++) {
+			if (filter == CF_NONE) {
+				return percentage;
+			} else if (filter == CF_ANY) {
+				int transported = GetCargoTransportedPercentsIfValid(i, id);
+				if (transported != -1) produced_cargo_count++;
+				percentage += transported == -1 ? 0 : transported;
```

Comment:
Should consolidate the `transported` checks
```suggestion
				if (transported != -1) {
					produced_cargo_count++;
					percentage += transported;
				}
```

Reply:
Thanks. it is done.